fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/logging/KeyValueLogMessage.java

```diff
@@ -37,40 +37,50 @@
  */
 @API(API.Status.MAINTAINED)
 public class KeyValueLogMessage {
+    @Nonnull
     private final long timestamp;
+    @Nonnull
     private final String staticMessage;
 
+    @Nonnull
     private final Map<String, String> keyValueMap;
 
     public static String of(@Nonnull final String staticMessage, @Nullable final Object... keysAndValues) {
-        return new KeyValueLogMessage(staticMessage, keysAndValues).toString();
+        return new KeyValueLogMessage(staticMessage, mapFromKeysAndValues(keysAndValues)).toString();
     }
 
     public static KeyValueLogMessage build(@Nonnull final String staticMessage, @Nullable final Object... keysAndValues) {
-        return new KeyValueLogMessage(staticMessage, keysAndValues);
+        return new KeyValueLogMessage(staticMessage, mapFromKeysAndValues(keysAndValues));
     }
 
-    private KeyValueLogMessage(@Nonnull final String staticMessage, @Nullable final Object[] keysAndValues) {
-        this.staticMessage = staticMessage;
-        this.timestamp = System.currentTimeMillis();
-        keyValueMap = new TreeMap<>();
+    private static Map<String, String> mapFromKeysAndValues(@Nullable Object[] keysAndValues) {
+        Map<String, String> keyValueMap = new TreeMap<>();
         if (keysAndValues != null) {
             for (int i = 0; i < (keysAndValues.length / 2) * 2; i += 2) {
-                addKeyValueImpl(keysAndValues[i + 0], keysAndValues[i + 1]);
+                addKeyValueImpl(keyValueMap, keysAndValues[i], keysAndValues[i + 1]);
             }
             if (keysAndValues.length % 2 == 1) {
                 throw new IllegalArgumentException("keys and values don't match");
             }
         }
+        return keyValueMap;
     }
 
-    private void addKeyValueImpl(@Nullable final Object key, @Nullable Object value) {
+    private KeyValueLogMessage(@Nonnull final String staticMessage, @Nonnull Map<String, String> keyValueMap) {
+        this.staticMessage = staticMessage;
+        this.timestamp = System.currentTimeMillis();
+        this.keyValueMap = keyValueMap;
+    }
+
+    private static void addKeyValueImpl(@Nonnull Map<String, String> keyValueMap, @Nullable final Object key, @Nullable Object value) {
         if (key == null) {
             throw new IllegalArgumentException("null key passed to KeyValueLogMessage"); // better to throw an exception rather than silently swallow the key
```

Comment:
This exception has been removed from the constructor and put into a helper method that is used by the static initializer. This ensures that the `finalizer` never sees an incomplete object, which is what SpotBugs was concerned about. There are enough classes with this problem that I just suppressed the error. For some of them, we could consider doing this kind of change (or marking the class as `final`), but the story gets more complicated with others.